File: android/src/main/java/org/maplibre/reactnative/components/layers/types/symbol/MLRNSymbolLayerManager.java
package org.maplibre.reactnative.components.layers.types.symbol;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager;

public class MLRNSymbolLayerManager extends MLRNVectorLayerManager<MLRNSymbolLayer> {
    public static final String REACT_CLASS = "MLRNSymbolLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNSymbolLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNSymbolLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/symbol/MLRNSymbolLayerManager.kt
package org.maplibre.reactnative.components.layers.types.symbol

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager

class MLRNSymbolLayerManager : MLRNVectorLayerManager<MLRNSymbolLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNSymbolLayer {
        return MLRNSymbolLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNSymbolLayer"
    }
}
